doc/tutorial/alignment_file/alignment_file_snippets.cpp

```diff
@@ -0,0 +1,108 @@
+//![include]
+#include <seqan3/io/alignment_file/all.hpp>
+//![include]
+//![include_aligned_sequence]
+#include <seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp>
+//![include_aligned_sequence]
+
+using namespace seqan3;
+
+struct write_file_dummy_struct
+{
+    write_file_dummy_struct()
+    {
+
+auto file_raw = R"//![sam_file](
+@HD	VN:1.6	SO:coordinate
+@SQ	SN:ref	LN:45
+r001	99	ref	7	30	8M2I4M1D3M	=	37	39	TTAGATAAAGGATACTG	*
+r003	0	ref	9	30	5S6M	*	0	0	GCCTAAGCTAA	*
+r004	0	ref	16	30	6M14N5M	*	0	0	ATAGCTTCAGC	*
+r003	2064	ref	29	17	5M	*	0	0	TAGGC	*
+r001	147	ref	37	30	9M	=	7	-39	CAGCGGCAT	*	NM:i:1
+)//![sam_file]";
+
+        std::ofstream file{std::filesystem::temp_directory_path()/"example.sam"};
+        std::string str{file_raw};
+        file << str.substr(1); // skip first newline
+    }
+};
+
+write_file_dummy_struct go{};
+
+int main()
+{
+
+{
+//![writing]
+alignment_file_output fout{"/tmp/my.sam", fields<field::FLAG, field::MAPQ>{}};
+
+size_t mymapq{60};
+uint8_t flag{0};
+
+// ...
+
+fout.emplace_back(flag, mymapq);
+// or:
+fout.push_back(std::tie(flag, mymapq));
+//![writing]
+}
+
+{
+//![file_extensions]
+debug_stream << format_sam::file_extensions << std::endl; // prints [fastq,fq]
+format_sam::file_extensions.push_back("sm");
+//![file_extensions]
+}
+
+{
+//![filename_construction]
+alignment_file_input fin{"/tmp/my.sam"};
+//![filename_construction]
+}
+
+/*
+//![construct_from_cin]
+alignment_file_input fin{std::cin, format_sam{}};
+//![construct_from_cin]
+*/
+
+{
+//![read_custom_fields]
+alignment_file_input fin{"/tmp/my.sam", fields<field::ID, field::SEQ, field::FLAG>{}};
+
+for (auto & [id, seq, flag /*order!*/] : fin)
+{
+    debug_stream << id << std::endl;
+    debug_stream << seq << std::endl;
+    debug_stream << flag << std::endl;
+}
+//![read_custom_fields]
+}
+
+{
+//![alignments_without_ref]
+alignment_file_input fin{"/tmp/my.sam", fields<field::ID, field::ALIGNMENT>{}};
+
+for (auto & [ id, alignment ] : fin)
+{
+    debug_stream << id << ": " << get<1>(alignment) << std::endl;
+}
+//![alignments_without_ref]
+}
+
+{
+//![alignments_with_ref]
+std::vector<std::string> ref_ids{"ref"}; // list of one reference name
+std::vector<dna5_vector> ref_sequences{"AGAGTTCGAGATCGAGGACTAGCGACGAGGCAGCGAGCGATCGAT"_dna5};
+
+alignment_file_input fin{"/tmp/example.sam", ref_ids, ref_sequences, fields<field::ALIGNMENT>{}};
+
+for (auto & [ alignment ] : fin)
+{
+    debug_stream << alignment << std::endl; // Now you can print the whole alignment!
+}
+//![alignments_with_ref]
+}
+
+}
```

Comment:
Hm, I thought we said every snippet should be self-contained and copy'n'pastable?

Reply:
Well at the top of the tutorial I state `include this file` and then all snippets are copy'n'pastable right now. Is this okay? (there are no missing variables or includes, only the main and the inlcude of alignment_file/all.hpp) I thought that this is fine then since every c++ developer should be able to wrote a main function?